Sources/GeneratorCLI/GeneratorCLI.swift

```diff
@@ -100,35 +100,31 @@ extension GeneratorCLI {
 
     @Option(
       help: """
-      CPU architecture of the host triple of the bundle. Defaults to a triple of the machine this generator is \
-      running on if unspecified. Available options: \(
-        Triple.Arch.allCases.map { "`\($0.rawValue)`" }.joined(separator: ", ")
-      ).
+      The host triple of the bundle. Defaults to a triple of the machine this generator is \
+      running on if unspecified.
       """
     )
-    var hostArch: Triple.Arch? = nil
+    var host: Triple? = nil
 
     @Option(
       help: """
-      CPU architecture of the target triple of the bundle. Same as the host triple CPU architecture if unspecified. \
-      Available options: \(Triple.Arch.allCases.map { "`\($0.rawValue)`" }.joined(separator: ", ")).
+      The target triple of the bundle. The default depends on a recipe used for SDK generation. Pass `--help` to a specific recipe subcommand for more details.
       """
     )
-    var targetArch: Triple.Arch? = nil
+    var target: Triple? = nil
```

Comment:
Do we need to keep previous option names for compatibility? I'm not sure if @euanh is strictly relying on the current spelling in CLI invocations or on the higher level API.

Reply:
@kateinoigakukun I guess if it's invoked as `swift-sdk-generator make-linux-sdk` nothing would break, right?

Reply:
It looks like the `--target/--host` triple arguments are in the top level `swift-sdk-generator` call.  Maybe they should be delegated to the subcommand?
Alternatively, if we have the full `--target` triple, we have all the information we need to dispatch to Linux or WASI SDK generation, so maybe we don't need the `make-linux-sdk/make-wasi-sdk` commands at all?

Reply:
I'm ok to keep that compatibility 👍 

@MaxDesiatov 
> I guess if it's invoked as swift-sdk-generator make-linux-sdk nothing would break, right?

Ah, good idea. Currently `swift-sdk-generator` is just an alias of `swift-sdk-generator make-linux-sdk`, so they both don't have compatibility with old one. But we can keep compatibility only for the non-subcommand version by having `GeneratorCLI.run()` separately.

@euanh 
> It looks like the --target/--host triple arguments are in the top level swift-sdk-generator call. Maybe they should be delegated to the subcommand?

The `GeneratorOptions` that `--target/--host` belongs to is embedded for each subcommand with `@OptionGroup`, so it's already delegated to subcommand.


> Alternatively, if we have the full --target triple, we have all the information we need to dispatch to Linux or WASI SDK generation, so maybe we don't need the make-linux-sdk/make-wasi-sdk commands at all?

We still need those subcommand separation to have different set of options for each recipe subcommand

Reply:
I don't care too much for compatibility at this point since we haven't tagged a single version yet. But I want to make sure these changes don't breaking anything for Euan unnecessarily, or at least we communicate them clearly and there's a clear migration path.

Reply:
Makes sense to me. Updated to keep compatibility while emitting deprecation warning

Reply:
@kateinoigakukun Could you update the PR title to reflect that it is just deprecating the old options?
I tweaked it slightly because it used to say `--target-cpu` but the flag is actually `--target-arch`.